services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRedirectUrisEnforcerExecutor.java

```diff
@@ -205,12 +193,34 @@ public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyExcep
                 if (client == null) {
                     throw invalidRedirectUri("Invalid parameter: clientId");
                 }
-                verifyRedirectUri(redirectUriParam, true);
+                if (isAuthFlowWithRedirectEnabled(client)) {
+                    verifyRedirectUri(redirectUriParam, true);
+                }
                 return;
             default:
         }
     }
 
+    private void verifyRedirectUris(ClientCRUDContext context) throws ClientPolicyException {
+        ClientRepresentation client = context.getProposedClientRepresentation();
+        if (isAuthFlowWithRedirectEnabled(client)) {
+            List<String> redirectUris = client.getRedirectUris();
+            if (redirectUris == null || redirectUris.isEmpty()) {
+                throw invalidRedirectUri(ERR_GENERAL);
+            }
+            verifyRedirectUris(client.getRootUrl(), redirectUris);
+            verifyPostLogoutRedirectUriUpdate(client);
+        }
+    }
+
+    private boolean isAuthFlowWithRedirectEnabled(ClientModel client) {
+        return client.isStandardFlowEnabled() || client.isImplicitFlowEnabled();
+    }
+
+    private static boolean isAuthFlowWithRedirectEnabled(ClientRepresentation client) {
+        return client.isStandardFlowEnabled() == Boolean.TRUE || client.isImplicitFlowEnabled() == Boolean.TRUE;
```

Comment:
Is this check correct in case any of the properties returns `null`? What are the default values on the ClientModel?

Reply:
Would it make sense to create an instance of a `ClientModel` and pass it to that function so that I can evaluate the outcome based on the model instead of the representation?
This way i would not need to make an assumption about the default values here.